File: iCopySharp/Classes/Utilities.cs
using System;
using System.IO;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace iCopy
{
    static class Utilities
    {
        public static MsgBoxResult MsgBoxWrap(string Message, MsgBoxStyle Style = MsgBoxStyle.DefaultButton1, string Title = "iCopy")
        {
            if (Properties.Settings.Default.Silent == Conversions.ToBoolean("False"))
            {
                return Interaction.MsgBox(Message, Style, Title);
            }
            else
            {
                Console.WriteLine(Message);
                return MsgBoxResult.Cancel;
            }
        }

        public static string GetWritablePath()
        {
            try
            {
                var fi = new FileInfo(Application.ExecutablePath);
                string path = fi.DirectoryName;

                var fs = File.Create(path + @"\writable");
                fs.Close();
                File.Delete(path + @"\writable");
                return path;
            }
            catch (Exception ex)
            {
                string path = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
                path = Path.Combine(path, Application.ProductName);
                if (!Directory.Exists(path))
                {
                    try
                    {
                        Directory.CreateDirectory(path);
                        Directory.Delete(path);
                    }
                    catch (Exception e)
                    {
                        throw;
                    }
                }
                return path;

            }

            return Application.LocalUserAppDataPath;
        }
    }
}

File: iCopySharp/Classes/Utilities.vb
Imports System.IO
Module Utilities
    Function MsgBoxWrap(Message As String, Optional Style As MsgBoxStyle = MsgBoxStyle.DefaultButton1, Optional Title As String = "iCopy") As MsgBoxResult
        If My.Settings.Silent = "False" Then
            Return MsgBox(Message, Style, Title)
        Else
            Console.WriteLine(Message)
            Return MsgBoxResult.Cancel
        End If
    End Function

    Function GetWritablePath() As String
        Try
            Dim fi As New System.IO.FileInfo(Application.ExecutablePath)
            Dim path As String = fi.DirectoryName

            Dim fs As FileStream = File.Create(path + "\writable")
            fs.Close()
            File.Delete(path + "\writable")
            Return path
        Catch ex As Exception
            Dim path As String = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)
            path = IO.Path.Combine(path, Application.ProductName)
            If Not Directory.Exists(path) Then
                Try
                    Directory.CreateDirectory(path)
                    Directory.Delete(path)
                Catch e As Exception
                    Throw
                End Try
            End If
            Return path

        End Try

        Return Application.LocalUserAppDataPath
    End Function
End Module
